### DISCUSSION TITLE
Get data from server only on first page load

### DISCUSSION THREAD

**Discussion by Aniket-IN** [inertiajs/inertia#1107] [Help] [answered]
Hi guys,
I'm very new to Inertia, 
My Stack: `laravel`, `Inertia`, `React`.

I have got a few data that needs to be included on every page, 

I tried using `HandleInertiaRequests` Middleware 
but it fetches the data from the database on every Inertia Request (which I don't want).

What I want is very similar to `Inertia::lazy( ... )`, but the only difference is I want the data to load on the first page by default.
Then maybe I can store that data to `localstorage` or something like that.

**Comment by ajnsn** [comment, marked as answer]
Hey @Aniket-IN 
you can check for Inertia's Header - it will only be there on an Inertia vist. Initial visits do not contain the header. This way you can achieve the desired "load once" behaviour, e.g. 

```php
use Illuminate\Http\Request;
use Inertia\Inertia;
use Inertia\Response;

class ExampleController
{
    public function __invoke(Request $request): Response
    {
        return Inertia::render('Example', [
            'onlyOnInitialVisit' => !$request->inertia() ? 'Hello World' : null,
        ]);
    }
}
```

**Reply by Aniket-IN** [reply]
You are awesome @ajnsn, 
This is exactly what I was looking for.

Now on the first load after getting the data from the server, I'm storing it in `localStorage` and whenever the server sends the data again I'm overwriting the `localStorage` :)

Thank you so much :)
